doc/api/timers.md

```diff
@@ -151,8 +151,9 @@ next event loop iteration.
 
 If `callback` is not a function, a [`TypeError`][] will be thrown.
 
-This method has a custom variant for promises that is available using
-[`util.promisify()`][]:
+If this method is invoked as its [`util.promisify()`][]ed version, it returns a
+Promise. The created `Immediate` is attached to the Promise as a `immediate`
```

Comment:
this seems like a huge problem. Promises are intended to be safely immutable - in that, you can always hand a Promise out to someone, and you give them no control over your conceptual Task (the thing that created the Promise).

This breaks that - and since `Promise.resolve(promise) === promise`, there's no obvious way to protect yourself from this hazard.

Reply:
My only justification is that this pattern has been implemented in child_process, 
which I admit is weak. 

modulo discussion of collaboration with whatwg would you accept a pull request that adds a function to the `timers` module?

```
interface Options<T> {
  signal: AbortSignal;
  values: T;
}
function delay<T>(n: number, options: Options<T>): Promise<T>
```

Reply:
I also think that `child_process`s should be thenable rather that doing something like this, fwiw.

Reply:
I agree this is confusing, worth mentioning that `Promise.resolve(foo)` happens in a ton of places - like returning from async functions:

```js
async function wait5000() {
  const immediate = setTimeout(5000);
  return immediate;
}
clearTimeout(wait5000()); // does not work
```